### ISSUE TITLE
math functions conflicts with types

### ISSUE THREAD

**Issue by HellGate94** [Unity-Technologies/Unity.Mathematics#190]
using static Unity.Mathematics.math; will cause conflicts with types like float3 type, preventing you from using float3.zero ('math.float3(float, float, float)' is a method, which is not valid in the given context)

naming of those functions should be reconsidered

**Comment by MrUnbelievable92** [comment]
Same goes for, for instance, `quaternion.Euler(float3 xyz)` - really annoying. This forces one to prefix all functions with `math.`.

**Comment by unpacklo** [comment]
Yeah, it's an unfortunate mistake we made a long time ago. I plan to fix this in the future when we can do a major API breaking change.

**Comment by visose** [comment]
Is the mistake using static method to avoid the new keyword?

**Comment by unpacklo** [comment]
> Is the mistake using static methods to avoid the new keyword?

Basically, yes. The intent was to be able to be as close to hlsl syntax as possible so you wouldn't have to insert `new` everywhere when you're trying to initialize a type, but we overlooked the fact `using static Unity.Mathematics.math;` would cause problems elsewhere:

<img width="740" alt="image" src="https://user-images.githubusercontent.com/3121968/115450857-4737ad00-a1d1-11eb-81fd-0aee580eccf4.png">

**Comment by Peaj** [comment]
Are there any news on this?
I am curious what the solution might be besides just not using `using static Unity.Mathematics.math;` or removing the static initializers and using new everywhere.
I really love the hlsl like syntax but this problem is a showstopper ☹️
